### DISCUSSION TITLE
sin(pi) and cos(pi/2) not zero

### DISCUSSION THREAD

**Discussion by ybh10** [josdejong/mathjs#2928] [Q&A]
This problem was discussed here: https://github.com/josdejong/mathjs/issues/447
The solution to use big numbers does not seem to work anymore.
`math.config({number: 'BigNumber'});`
`math.evaluate('cos(pi/2)').toString() //=1.539082031431044993140174126710585339910740432566411533235469223e-64 != 0`
Is there a way to do it now with a full expression?

**Comment by josdejong** [comment]
Both `number` and `BigNumber` are floating point numbers, so you can easily get round-off errors.

For displaying to a user, the easiest is to round the number to a limited number of digits.

Here some documentation: https://mathjs.org/docs/datatypes/numbers.html#roundoff-errors

**Reply by kgryte** [reply]
There are ways of handling this without requiring manual rounding in userland. Namely, dedicated APIs supporting [argument reduction](https://github.com/stdlib-js/stdlib/tree/ede9ef830c3e6efcdf0f427a1dc651856c41e6ab/lib/node_modules/%40stdlib/math/base/special/cospi) which allow for more accurate computation for larger inputs. One could re-implement in Math.js, or just rely on, e.g., stdlib, to address the above use case.
